FarmSimulator1/src/Domain/Product.java

```diff
@@ -0,0 +1,94 @@
+package Domain;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+/**
+ * The Product class is the main class for our shop. She is in charge for all the products that are added in our shop.
+ * She contain the atributes for our products.
+ */
+public abstract class Product implements Comparable{
+    Integer id;
+    private double price;
+    private LocalDate validity_date;
+    private double weight;
+    private String name;
+
+
+    public Product(int id, double price, LocalDate validity_date, double weight, String name) {
```

Comment:
if your class property is Integer, why make it int here? it's unnecessary boxing/unboxing